src/Simplify_Div.cpp

```diff
@@ -3,112 +3,45 @@
-        // take the intersection with that.
-        if (a_bounds.max_defined && a_bounds.min_defined) {
-            int64_t v = std::max(a_bounds.max, -a_bounds.min);
-            if (bounds->min_defined) {
-                bounds->min = std::max(bounds->min, -v);
-            } else {
-                bounds->min = -v;
-            }
-            if (bounds->max_defined) {
-                bounds->max = std::min(bounds->max, v);
-            } else {
-                bounds->max = v;
-            }
-            bounds->min_defined = bounds->max_defined = true;
-        }
-
-        // Bounded numerator divided by constantish
-        // denominator can sometimes collapse things to a
-        // constant at this point
-        if (bounds->min_defined &&
-            bounds->max_defined &&
-            bounds->max == bounds->min) {
-            if (op->type.can_represent(bounds->min)) {
-                return make_const(op->type, bounds->min);
-            } else {
-                // Even though this is 'no-overflow-int', if the result
-                // we calculate can't fit into the destination type,
-                // we're better off returning an overflow condition than
-                // a known-wrong value. (Note that no_overflow_int() should
-                // only be true for signed integers.)
-                internal_assert(op->type.is_int());
-                clear_bounds_info(bounds);
-                return make_signed_integer_overflow(op->type);
+Expr Simplify::visit(const Div *op, ExprInfo *info) {
+    ExprInfo a_info, b_info;
+    Expr a = mutate(op->a, &a_info);
+    Expr b = mutate(op->b, &b_info);
+
+    if (info) {
+        if (op->type.is_int_or_uint()) {
+            info->bounds = a_info.bounds / b_info.bounds;
+            info->alignment = a_info.alignment / b_info.alignment;
+            info->trim_bounds_using_alignment();
+            info->cast_to(op->type);
+
+            // Bounded numerator divided by constantish
+            // denominator can sometimes collapse things to a
+            // constant at this point
+            if (info->bounds.is_single_point()) {
+                if (op->type.can_represent(info->bounds.min)) {
+                    return make_const(op->type, info->bounds.min);
+                } else {
+                    // Even though this is 'no-overflow-int', if the result
+                    // we calculate can't fit into the destination type,
+                    // we're better off returning an overflow condition than
+                    // a known-wrong value. (Note that no_overflow_int() should
+                    // only be true for signed integers.)
+                    internal_assert(no_overflow_int(op->type));
+                    clear_expr_info(info);
+                    return make_signed_integer_overflow(op->type);
+                }
             }
+        } else {
+            // TODO: division bounds for floating-point division
```

Comment:
Does this warrant an open issue? We didn't previously support bounds on floating-point division, right?

Reply:
That's correct. Main only supports int32/64 constant bounds.

I don't think it's worth an issue. This is not something I'd suggest someone to work on. It's more of a placeholder to say "hey, if we need this, here's where to put the logic"